sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/FileWriterFactory.scala

```diff
@@ -47,9 +47,12 @@ case class FileWriterFactory (
     }
   }
 
-  private def createTaskAttemptContext(partitionId: Int): TaskAttemptContextImpl = {
+  // VisibleForTesting
+  private[v2] def createTaskAttemptContext(
```

Comment:
can we use `PrivateMethodTester` in `FileWriterFactorySuite` to avoid expanding the scope of this function

Reply:
done